src/main/scala/org/broadinstitute/dsde/workbench/avram/db/CollectionComponent.scala

```diff
@@ -0,0 +1,65 @@
+package org.broadinstitute.dsde.workbench.avram.db
+
+import java.sql.Timestamp
+import java.time.Instant
+
+import org.broadinstitute.dsde.workbench.avram.api.Collection
+import AvramPostgresProfile.api._
+
+import scala.concurrent.ExecutionContext
+
+case class CollectionRecord(id: Long,
+                            name: String,
+                            createdBy: String,
+                            samResource: String,
```

Comment:
I was going to suppress this nit but it's actually a bug. The order of the fields here is different than the order of the columns in `CollectionTable`. The * projection uses the order from `CollectionTable` which means that the `createdBy` and `samResource` values are being swapped... and they're both `String` so there's no compilation error... and the technique of defining the * projection uses `CollectionRecord.tupled` so there's no way to avoid the problem by using named parameters.

Short term, we need to make sure the fields are consistently listed in the same order. Long term, I'd like to figure out how to craft the code to make these kinds of errors impossible (or at least more obvious).

Reply:
Good catch!